server/src/test/java/org/elasticsearch/cluster/metadata/TemplateUpgradeServiceTests.java

```diff
@@ -75,8 +81,25 @@
 
 public class TemplateUpgradeServiceTests extends ESTestCase {
 
-    private final ClusterService clusterService = new ClusterService(Settings.EMPTY, new ClusterSettings(Settings.EMPTY,
-        ClusterSettings.BUILT_IN_CLUSTER_SETTINGS), null, Collections.emptyMap());
+    private ThreadPool threadPool;
+    private ClusterService clusterService;
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        threadPool = new TestThreadPool("TemplateUpgradeServiceTests");
+        clusterService = createClusterService(threadPool);
+    }
+
+    @Override
+    @After
+    public void tearDown() throws Exception {
+        ThreadPool.terminate(threadPool, 30, TimeUnit.SECONDS);
```

Comment:
we can just call `terminate(threadPool)` here

Reply:
I have called `shutdownNow()`, all threads should terminate, unless they are chomping `InterruptedException`.